src/index.js

```diff
@@ -16,5 +16,64 @@ module.exports = {
    * This gives you an opportunity to set up your data model,
    * run jobs, or perform some special logic.
    */
-  bootstrap(/*{ strapi }*/) {},
-};
+  bootstrap({ strapi }) {
+
+    strapi.db.lifecycles.subscribe({
+      models: ["plugin::users-permissions.user"],
+      // afterCreate lifecycle to create a new team-membership entry when a new user is created
+      async afterCreate({result}){
+        const {projects} = result
+
+        for (const project of projects) {
+          await strapi.service("api::team-membership.team-membership").create({
+            data: { 
+              user: [result.id],
+              project: [project.id],
```

Comment:
What if the user hasn't join a project? We shouldn't create an entry in team-membership if the user hasn't join a project,

Reply:
The` afterCreate` lifecycle triggers a `For loop` which loops over a list of projects the new `user` is a part of. If the project list is empty, the `strapi.service(...).create(...)` will not create a new entry in the `team-membership`